ArgusCore/src/main/java/com/salesforce/dva/argus/entity/NumberOperations.java

```
@@ -0,0 +1,803 @@
+			case LONG:
+				return Math.sqrt(n.longValue());
+			case DOUBLE:
+				return Math.sqrt(n.doubleValue());
+			default:
+				throw new IllegalArgumentException();
+		}
+	}
+	
+	/**
+	 * @param n the Number object to test
+	 * @return true iff n is not a number (e.g. Double.NaN, imaginary numbers)
+	 */
+	public static boolean isNaN(Number n) {
+		switch (ValueType.value(n)) {
+			case INT:
+				return false;
+			case LONG:
+				return false;
+			case DOUBLE:
+				return Double.isNaN(n.doubleValue());
+			default:
+				throw new IllegalArgumentException();
+		}
+	}
+	
+	/**
+	 * @param vals the list of values to find the minimum of. Must be nonempty and contain no null elements.
+	 * @return the minimum element in vals.
+	 */
+	public static Number min(List<Number> vals) {
+		return Collections.min(vals, new Comparator<Number>() {
+
+			@Override
+			public int compare(Number o1, Number o2) {
+				return NumberOperations.compare(o1, o2);
+			}
+			
+		});
+	}
+	
+	/**
+	 * @param vals the list of values to find the maximum of. Must be nonempty and contain no null elements.
+	 * @return the maximum element in vals.
+	 */
+	public static Number max(List<Number> vals) {
+		return Collections.max(vals, new Comparator<Number>() {
+
+			@Override
+			public int compare(Number o1, Number o2) {
+				return NumberOperations.compare(o1, o2);
+			}
+			
+		});
+	}
+	
+	/**
+	 * @param vals the numbers to sum. Must contain no null elements.
+	 * @return the sum of all elements in vals.
+	 */
+	public static Number sum(List<Number> vals) {
+		return vals.stream()
+				.reduce(0, (a, b) -> add(a, b));
+	}
+	
+	/**
+	 * @param vals the list of numbers
+	 * @return the list of doubles contained in vals. If one element of vals is null, will preserve it as null.
+	 * @throws IllegalArgumentException if one of the numbers in vals is not a double
+	 */
+	public static List<Double> getListAsDoubles(List<Number> vals) throws IllegalArgumentException {
```

`IllegalArgumentException` is unchecked, so I believe there's no need to indicate it in the method signature; javadoc is good enough. 
Also, from a quick Command+F it seems this method is always used inside a `try-catch` block. It seems cleaner to just let the exception be thrown at this level with a message like `Expected class type of List elements to be Doubles` and have it be called without try-catch; the stack trace will say what transform it failed on.